app/Http/Controllers/ProfileController.php

```diff
@@ -135,4 +135,95 @@ public function mobileCarrier(Request $request){
             return (new ErrorResource($res))->response()->setStatusCode(400);
         }
     }
+
+    public function updateProfile(Request $request, Validate $validate){ 
+      
+       $user = $request->user();
+       $input = $request->all();
+       $validationErrors = $validate->validate($request, $this->rules->getUpdateProfileValidatonRules(),[]);
+       if( $validationErrors ){
+           return (new ErrorResource($validationErrors))->response()->setStatusCode(400);
+       }
+
+       try{           
+            $private_flags = [];
+            if(isset($input['first_name']))
+                $user->first_name =  $input['first_name']; 
+            if(isset($input['middle_name']))
+                $user->middle_name = $input['middle_name'] ;
+            if(isset($input['last_name']))
+                $user->last_name =  $input['last_name'];
+            if(isset($input['address_1']))
+                $user->address_1 = $input['address_1'];
+            if(isset($input['address_2']))
+                $user->address_2 = $input['address_2']; 
+            if(isset($input['city']))
+                $user->city = $input['city'];
+            if(isset($input['state']))
+                $user->state =  $input['state'];
+            if(isset($input['country']))
+                $user->country = $input['country'];
+            if(isset($input['postal_code']))
+                $user->postal_code = $input['postal_code']; 
+            if(isset($input['gender']))
+                $user->gender = $input['gender'];
+            if(isset($input['birthday']))
+                $user->birthday = date('Y-m-d', strtotime($input['birthday']));
+            if(isset($input['language']))
+                $user->language = $input['language']; 
+            if(isset($input['phone_number']))
+                $user->phone_number = $input['phone_number'];
+            if(isset($input['mobile_carrier']))
+                $user->phone_number = $input['mobile_carrier'];
+            if(isset($input['default_algo']))
+                $user->default_algo = $input['default_algo'];
```

Comment:
You need use `nullable` or `sometimes` in the rules validation for optional parameters